src/main/java/org/ohdsi/webapi/statistic/controller/StatisticController.java

```
@@ -0,0 +1,255 @@
+package org.ohdsi.webapi.statistic.controller;
+
+import com.opencsv.CSVWriter;
+
+import org.ohdsi.webapi.statistic.dto.AccessTrendDto;
+import org.ohdsi.webapi.statistic.dto.AccessTrendsDto;
+import org.ohdsi.webapi.statistic.dto.EndpointDto;
+import org.ohdsi.webapi.statistic.dto.SourceExecutionDto;
+import org.ohdsi.webapi.statistic.dto.SourceExecutionsDto;
+import org.ohdsi.webapi.statistic.service.StatisticService;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Controller;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import java.io.ByteArrayOutputStream;
+import java.io.StringWriter;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Controller
+@Path("/statistic/")
+@ConditionalOnProperty(value = "audit.trail.enabled", havingValue = "true")
```

If Audit Trail feature is not enabled the Controller is not efficient

---

Is the intent here to make the `/statistic/` end-point available based on the `audit.trail.enabled` property? Seems a little confusing to make an endpoint conditional vs. having it always available and to have it return based on the status of the configuration of the application.

---

I personally see no problem that some of the endpoints are not available if an appropriate configuration is not at place

---

I agree with @anthonysena, it is confusing because people will look at api docs and think they can call an endpoint, but it won't exist, which is confusing.

I don't see any documentation that describes that this end-piont doesn't exist when the configuration is not set.

We have rules about opt-in flags for default behavior (such as logging should be turned on/off, or auditing is enabled/distables) but the invocation of a end-point is the opt-in request of the user.   If it's not enabled, it would be better to raise an error message saying 'auditing is not enabled' then return a 404 that a conditional-endpoint does not exist.

---

Changed as suggested